base/server/src/main/java/com/netscape/cms/realm/PKILDAPRealm.java

```diff
@@ -0,0 +1,415 @@
+                switch(name) {
+                case "usersDN":
+                case "groupsDN":
+                    break;
+                case "url":
+                    logger.info("- URL: " + value);
+
+                    URI url;
+                    try {
+                        url = new URI(value);
+                    } catch (URISyntaxException e) {
+                        throw new LifecycleException("No valid ldap url : " + value);
+                    }
+                    String host = url.getHost();
+                    String port = "" + url.getPort();
+
+                    String protocol = url.getScheme();
+                    String secureConn;
+                    if ("ldap".equals(protocol)) {
+                        secureConn = "false";
+                    } else if ("ldaps".equals(protocol)) {
+                        secureConn = "true";
+                    } else {
+                        throw new LifecycleException("Unsupported LDAP protocol: " + protocol);
+                    }
+
+                    ldapConfig.put("ldapconn.host", host);
+                    ldapConfig.put("ldapconn.port", port);
+                    ldapConfig.put("ldapconn.secureConn", secureConn);
+                    break;
+                case "authType":
+                    logger.info("- authentication type: " + value);
+                    ldapConfig.put("ldapauth.authtype", value);
+                    break;
+                case "bindDN":
+                    logger.info("- bind DN: " + value);
+                    ldapConfig.put("ldapauth.bindDN", value);
+                    break;
+                case "bindPassword":
+                    ldapConfig.put("ldapauth.bindPassword", value);
+                    break;
+                case "nickname":
+                    logger.info("- nickname: " + value);
+                    ldapConfig.put("ldapauth.clientCertNickname", value);
+                    break;
+                case "minConns","maxConns","maxResults", "errorIfDown":
+                    logger.info("- " + name + ": " + value);
+                    ldapConfig.put(name, value);
+                    break;
+                default:
+                    if(name.startsWith("ldapauth.") || name.startsWith("ldapconn.")) {
+                        logger.info("- " + name + ": " + value);
+                        ldapConfig.put(name, value);
+                    }
+                    else {
+                        cs.put(name, value);
+                    }
+                }
+            }
+
+        } else {
+
+            logger.info("Loading LDAP realm config from " + configFile);
+            try {
+                cs = new EngineConfig(new FileConfigStorage(configFile));
+                cs.load();
+
+                ps = IPasswordStore.create(cs.getPasswordStoreConfig());
+            } catch (Exception e) {
+                throw new LifecycleException("Cannot load config file");
```

Comment:
Could you add the filename to the exception message, and attach the original exception too? That will help troubleshooting.